src/objectExplorer/objectExplorerProvider.ts

```diff
@@ -49,11 +49,11 @@ export class ObjectExplorerProvider implements vscode.TreeDataProvider<any> {
         return undefined;
     }
 
-    public removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
+    public async removeObjectExplorerNode(node: TreeNodeInfo, isDisconnect: boolean = false): Promise<void> {
         return this._objectExplorerService.removeObjectExplorerNode(node, isDisconnect);
     }
 
-    public refreshNode(node: TreeNodeInfo): Promise<void> {
+    public async refreshNode(node: TreeNodeInfo): Promise<void> {
         return this._objectExplorerService.refreshNode(node);
```

Comment:
So this is calling this._objectExplorerService.refreshNode which then calls return this._objectExplorerProvider.refresh(node)...that seems a bit unnecessary don't you think? Why aren't callers just calling _objectExplorerService.refreshNode themselves directly instead of getting routed through here first?

Reply:
Because you need to call it via the tree provider for the tree view to refresh